From 638effcf598a495b4928c0bc1da021021146e4f5 Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Tue, 26 Aug 2014 19:41:07 +0530
Subject: [PATCH] ASoC: Add kcontrol to soc_bytes_ext get and put

Issue: GMINL-7223

Add kcontrol to the get and put function pointers in
soc_bytes_ext, as it is needed for referencing the corresponding
algo control.

Change-Id: Ia88f75ccdda28ff4cb6e5728c1526f79ef1ac433
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 include/sound/soc.h  | 6 ++++--
 sound/soc/soc-core.c | 4 ++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/sound/soc.h b/include/sound/soc.h
index 4cb0ca9747c5..97fe9d96b3f1 100644
--- a/include/sound/soc.h
+++ b/include/sound/soc.h
@@ -889,8 +889,10 @@ struct soc_bytes_ext {
 	int index;
 
 	/* used for TLV byte control */
-	int (*get)(unsigned int __user *bytes, unsigned int size);
-	int (*put)(const unsigned int __user *bytes, unsigned int size);
+	int (*get)(struct snd_kcontrol *kcontrol,
+			unsigned int __user *bytes, unsigned int size);
+	int (*put)(struct snd_kcontrol *kcontrol,
+			const unsigned int __user *bytes, unsigned int size);
 
 	unsigned int pvt_data_len;
 	char *pvt_data;
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 5a430ac297b4..989761096292 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -3412,11 +3412,11 @@ int snd_soc_bytes_tlv_callback(struct snd_kcontrol *kcontrol, int op_flag,
 	switch (op_flag) {
 	case SNDRV_CTL_TLV_OP_READ:
 		if (params->get)
-			ret = params->get(tlv, count);
+			ret = params->get(kcontrol, tlv, count);
 		break;
 	case SNDRV_CTL_TLV_OP_WRITE:
 		if (params->put)
-			ret = params->put(tlv, count);
+			ret = params->put(kcontrol, tlv, count);
 		break;
 	}
 	return ret;
-- 
1.9.1

